vault/provider.go

```diff
@@ -705,6 +706,16 @@ func providerToken(d *schema.ResourceData) (string, error) {
 	return strings.TrimSpace(token), nil
 }
 
+func RecordMultipleStates(lastStates ...string) func(*api.Response) {
+	return func(resp *api.Response) {
+		var m sync.Mutex
+		newState := resp.Header.Get("X-Vault-Index")
+		m.Lock()
+		lastStates = api.MergeStates(lastStates, newState)
```

Comment:
I don't think this will actually update the `lastStates` slice in `providerConfigure`'s scope. Maybe worth adding a test to verify either way.

Reply:
Yeah, `lastState` will not have the updated value outside of this func (`RecordState ` accepts a string pointer which is why it works there).

Reply:
@calvn is there a way to make this work with a string slice? Or are string pointers our best bet here? (can we pass arrays by pointers in Go?)

Reply:
Though this is not common, we could have as the argument type `*[]string` and we pass in the pointer to the slice for `lastStates` whenever this is called.